File: 28. Implement strStr().py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        if not haystack and not needle:
            return 0
        for i in xrange(len(haystack)):
            if i + len(needle) <= len(haystack) and haystack[i:i + len(needle)] == needle:
                return i
        return -1

File: 28. Implement strStr().cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        if (haystack.size() == 0 && needle.size() == 0)
            return 0;
        for (int i = 0; i < haystack.size(); i++)
        {
            string temp = haystack.substr(i, needle.size());
            if (i + needle.size() <= haystack.size() && temp == needle)
                return i;
        }
        return -1;
    }
};
